public/Read-DbaBackupHeader.ps1

```diff
@@ -128,7 +128,15 @@ function Read-DbaBackupHeader {
             $restore = New-Object Microsoft.SqlServer.Management.Smo.Restore
 
             if ($DeviceType -eq 'URL') {
-                $restore.CredentialName = $AzureCredential
+                if (-not [String]::IsNullOrWhiteSpace($AzureCredential)) {
```

Comment:
The deal here is... Read-DbaBackupHeader is actually working right now. If you want to use a SAS URL, you need to remove the SAS from the URL and add the SAS credential in MSSQL. This is here for convenience, you either use an explicit credential or a SAS URL, and if it is a SAS url, MSSQL wants it without the SAS token.